bundles/org.openhab.binding.qolsysiq/src/main/java/org/openhab/binding/qolsysiq/internal/handler/QolsysIQZoneHandler.java

```diff
@@ -94,17 +94,38 @@ protected void updateZone(Zone zone) {
     }
 
     protected void zoneActiveEvent(ZoneActiveEvent event) {
-        if (event.zone.zoneId == zoneId()) {
+        if (event.zone.zoneId == getZoneId()) {
             updateZoneStatus(event.zone.status);
         }
     }
 
     protected void zoneUpdateEvent(ZoneUpdateEvent event) {
-        if (event.zone.zoneId == zoneId()) {
+        if (event.zone.zoneId == getZoneId()) {
             updateZone(event.zone);
         }
     }
 
+    private void initializeZone() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            BridgeHandler handler = bridge.getHandler();
+            if (handler instanceof QolsysIQPartitionHandler) {
+                if (handler.getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+                    return;
+                }
+                Zone z = ((QolsysIQPartitionHandler) handler).getZone(getZoneId());
+                if (z == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
```

Comment:
Please add a third argument to explain the configuration error to the user, something like "the zone setup does not match any existing zone".